src/installer.js

```diff
@@ -21,261 +21,231 @@ function defaultRename (dest, src) {
- */
-function createSpec (options, dir) {
-  const specSrc = path.resolve(__dirname, '../resources/spec.ejs')
-  const specDest = path.join(dir, 'nuget', `${options.name}.nuspec`)
-  options.logger(`Creating spec file at ${specDest}`)
-
-  return common.generateTemplate(options, specSrc)
-    .then(data => fs.outputFile(specDest, data))
-    .catch(common.wrapError('creating spec file'))
-}
+  /**
+   * Package everything using `nuget`.
+   */
+  createPackage () {
+    this.options.logger(`Creating package at ${this.stagingDir}`)
+
+    const cmd = path.join(this.vendorDir, 'nuget', 'nuget.exe')
+    const args = [
+      'pack',
+      this.specPath,
+      '-BasePath',
+      this.stagingAppDir,
+      '-OutputDirectory',
+      path.join(this.stagingDir, 'nuget'),
+      '-NoDefaultExcludes'
+    ]
+
+    return spawn(cmd, args, this.options.logger)
+      .catch(common.wrapError('creating package with NuGet'))
+  }
 
-/**
- * Copy the application into the package.
- */
-function createApplication (options, dir) {
-  const applicationDir = path.join(dir, options.name)
-  const updateSrc = path.resolve(__dirname, '../vendor/squirrel/Squirrel.exe')
-  const updateDest = path.join(applicationDir, 'Update.exe')
-  options.logger(`Copying application to ${applicationDir}`)
-
-  return fs.copy(options.src, applicationDir)
-    .then(() => fs.copy(updateSrc, updateDest))
-    .catch(common.wrapError('copying application directory'))
-}
+  /**
+   * Create the nuspec file for the package.
+   *
+   * See: https://docs.nuget.org/create/nuspec-reference
+   */
+  createSpec () {
+    const src = path.resolve(__dirname, '../resources/spec.ejs')
+    this.options.logger(`Creating spec file at ${this.specPath}`)
+
+    return this.createTemplatedFile(src, this.specPath)
+      .catch(common.wrapError('creating spec file'))
+  }
 
-/**
- * Create subdirectories where intermediate files will live.
- */
-function createSubdirs (options, dir) {
-  options.logger(`Creating subdirectories under ${dir}`)
+  /**
+   * Find the package just created.
+   */
+  findPackage () {
+    const packagePattern = path.join(this.stagingDir, 'nuget', '*.nupkg')
+    this.options.logger(`Finding package with pattern ${packagePattern}`)
 
-  return fs.ensureDir(path.join(dir, 'nuget'))
-    .then(() => fs.ensureDir(path.join(dir, 'squirrel')))
-    .then(() => dir)
-    .catch(common.wrapError('creating temporary subdirectories'))
-}
+    return glob(packagePattern)
+      .then(files => files[0])
+      .catch(common.wrapError('finding package with pattern'))
+  }
 
-/**
- * Create the contents of the package.
- */
-function createContents (options, dir) {
-  return common.createContents(options, dir, [
-    createSpec,
-    createApplication
-  ])
-}
+  /**
+   * Get the hash of default options for the installer. Some come from the info
+   * read from `package.json`, and some are hardcoded.
+   */
+  generateDefaults () {
+    return common.readMeta(this.userSupplied)
```

Comment:
This is the reason why it's failing on all CIs
```suggestion
    return common.readMetadata(this.userSupplied)
```

Reply:
Yup, fixed it elsewhere but forgot it was used here.